test/e2e_node/util.go

```diff
@@ -75,6 +75,9 @@ var startServices = flag.Bool("start-services", true, "If true, start local node
 var stopServices = flag.Bool("stop-services", true, "If true, stop local node services after running tests")
 var busyboxImage = imageutils.GetE2EImage(imageutils.BusyBox)
 
+// defaultMutableServiceGate is similar to DefaultMutableFeatureGate, but is intended for API server feature gates.
+var defaultMutableServiceGate featuregate.MutableFeatureGate = featuregate.NewFeatureGate()
```

Comment:
Why is is called `ServiceGate` instead of `FeatureGate`?

And why `default`?

What is this used for? If it's not used, remove it.

Reply:
Please see Sergey's comments above [[1](https://github.com/kubernetes/kubernetes/pull/120139#discussion_r1326349635)][[2](https://github.com/kubernetes/kubernetes/pull/120139#discussion_r1323685079)].

In short, as @SergeyKanzhelev said, the `serviceFeatureGates` argument "defines feature gates to set for the fake API server that will be served by the test".

> What is this used for?

To check whether service feature gates are enabled or not during tests.

> Why is is called ServiceGate instead of FeatureGate? And why default?

We can discuss the name. My reasoning was to follow `DefaultMutableFeatureGate`'s naming scheme, but for service feature gates. Here are some alternatives I can think about:
- mutableServiceGate
- defualtServiceGate
- serviceFeatureGate

Let me know if you have other suggestions.

> If it's not used, remove it.

Before this PR, checking whether a feature gate is enabled during tests was impossible, or at least extremely non-intuitive. Using `DefaultFeatureGate.Enabled()` did not work and always returned false since the variable was never properly initialized.

IMHO if a day comes where there would be a need to check if a service feature gate is enabled during tests it would be non-intuitive to figure out how to do so.

WDYT?

Reply:
+1 @pohly. cc @SergeyKanzhelev